src/battle_anim.c

```diff
@@ -0,0 +1,3348 @@
+    {gBattleAnimSpritePal_GlowyGreenOrb, ANIM_TAG_GLOWY_GREEN_ORB},
+    {gBattleAnimSpritePal_GreenSpike, ANIM_TAG_GREEN_SPIKE},
+    {gBattleAnimSpritePal_WhiteCircleOfLight, ANIM_TAG_WHITE_CIRCLE_OF_LIGHT},
+    {gBattleAnimSpritePal_GlowyBlueOrb, ANIM_TAG_GLOWY_BLUE_ORB},
+    {gBattleAnimSpritePal_Pokeblock, ANIM_TAG_POKEBLOCK},
+    {gBattleAnimSpritePal_WhiteFeather, ANIM_TAG_WHITE_FEATHER},
+    {gBattleAnimSpritePal_Sparkle6, ANIM_TAG_SPARKLE_6},
+    {gBattleAnimSpritePal_Splash, ANIM_TAG_SPLASH},
+    {gBattleAnimSpritePal_Splash, ANIM_TAG_SWEAT_BEAD},
+    {gBattleAnimSpritePal_Gem1, ANIM_TAG_GEM_1},
+    {gBattleAnimSpritePal_Gem1, ANIM_TAG_GEM_2},
+    {gBattleAnimSpritePal_Gem1, ANIM_TAG_GEM_3},
+    {gBattleAnimSpritePal_SlamHit2, ANIM_TAG_SLAM_HIT_2},
+    {gBattleAnimSpritePal_Recycle, ANIM_TAG_RECYCLE},
+    {gBattleAnimSpritePal_RedParticles, ANIM_TAG_RED_PARTICLES},
+    {gBattleAnimSpritePal_Protect, ANIM_TAG_PROTECT},
+    {gBattleAnimSpritePal_DirtMound, ANIM_TAG_DIRT_MOUND},
+    {gBattleAnimSpritePal_Shock3, ANIM_TAG_SHOCK_3},
+    {gBattleAnimSpritePal_WeatherBall, ANIM_TAG_WEATHER_BALL},
+    {gBattleAnimSpritePal_Bird, ANIM_TAG_BIRD},
+    {gBattleAnimSpritePal_CrossImpact, ANIM_TAG_CROSS_IMPACT},
+    {gBattleAnimSpritePal_Slash2, ANIM_TAG_SLASH_2},
+    {gBattleAnimSpritePal_WhipHit, ANIM_TAG_WHIP_HIT},
+    {gBattleAnimSpritePal_BlueRing2, ANIM_TAG_BLUE_RING_2},
+};
+
+const struct BattleAnimBackground gBattleAnimBackgroundTable[] =
+{
+    [BG_DARK_] = {gBattleAnimBgImage_Dark, gBattleAnimBgPalette_Dark, gBattleAnimBgTilemap_Dark},
+    [BG_DARK] = {gBattleAnimBgImage_Dark, gBattleAnimBgPalette_Dark, gBattleAnimBgTilemap_Dark},
+    [BG_GHOST] = {gBattleAnimBgImage_Ghost, gBattleAnimBgPalette_Ghost, gBattleAnimBgTilemap_Ghost},
+    [BG_PSYCHIC] = {gBattleAnimBgImage_Psychic, gBattleAnimBgPalette_Psychic, gBattleAnimBgTilemap_Psychic},
+    [BG_IMPACT_OPPONENT] = {gBattleAnimBgImage_Impact, gBattleAnimBgPalette_Impact, gBattleAnimBgTilemap_ImpactOpponent},
+    [BG_IMPACT_PLAYER] = {gBattleAnimBgImage_Impact, gBattleAnimBgPalette_Impact, gBattleAnimBgTilemap_ImpactPlayer},
+    [BG_IMPACT_CONTESTS] = {gBattleAnimBgImage_Impact, gBattleAnimBgPalette_Impact, gBattleAnimBgTilemap_ImpactContests},
+    [BG_DRILL] = {gBattleAnimBgImage_Drill, gBattleAnimBgPalette_Drill, gBattleAnimBgTilemap_Drill},
+    [BG_DRILL_CONTESTS] = {gBattleAnimBgImage_Drill, gBattleAnimBgPalette_Drill, gBattleAnimBgTilemap_DrillContests},
+    [BG_HIGHSPEED_OPPONENT] = {gBattleAnimBgImage_Highspeed, gBattleAnimBgPalette_Highspeed, gBattleAnimBgTilemap_HighspeedOpponent},
+    [BG_HIGHSPEED_PLAYER] = {gBattleAnimBgImage_Highspeed, gBattleAnimBgPalette_Highspeed, gBattleAnimBgTilemap_HighspeedPlayer},
+    [BG_THUNDER] = {gBattleAnimBgImage_Thunder, gBattleAnimBgPalette_Thunder, gBattleAnimBgTilemap_Thunder},
+    [BG_GUILLOTINE_OPPONENT] = {gBattleAnimBgImage_Guillotine, gBattleAnimBgPalette_Guillotine, gBattleAnimBgTilemap_Guilloti
```

Comment:
Are you sure you need this local variable?

Reply:
sadly, it looks that way